src/libraries/System.Net.Mail/tests/Functional/SmtpClientTest.cs

```diff
@@ -349,60 +345,38 @@ public void TestZeroTimeout()
         [InlineData(null)]
         public async Task TestMailDeliveryAsync(string body)
         {
-            SmtpServer server = new SmtpServer();
-            SmtpClient client = new SmtpClient("localhost", server.EndPoint.Port);
+            using var server = new MockSmtpServer();
+            using SmtpClient client = server.CreateClient();
             MailMessage msg = new MailMessage("foo@example.com", "bar@example.com", "hello", body);
-            string clientDomain = IPGlobalProperties.GetIPGlobalProperties().HostName.Trim().ToLower();
 
-            try
-            {
-                Thread t = new Thread(server.Run);
-                t.Start();
-                await client.SendMailAsync(msg).TimeoutAfter((int)TimeSpan.FromSeconds(30).TotalMilliseconds);
-                t.Join();
-
-                Assert.Equal("<foo@example.com>", server.MailFrom);
-                Assert.Equal("<bar@example.com>", server.MailTo);
-                Assert.Equal("hello", server.Subject);
-                Assert.Equal(body ?? "", server.Body);
-                Assert.Equal(clientDomain, server.ClientDomain);
-            }
-            finally
-            {
-                server.Stop();
-            }
+            await client.SendMailAsync(msg).TimeoutAfter((int)TimeSpan.FromSeconds(30).TotalMilliseconds);
+
+            Assert.Equal("<foo@example.com>", server.MailFrom);
+            Assert.Equal("<bar@example.com>", server.MailTo);
+            Assert.Equal("hello", server.Message.Subject);
+            Assert.Equal(body ?? "", server.Message.Body);
+            Assert.Equal(GetClientDomain(), server.ClientDomain);
         }
 
         [Fact]
+        [ActiveIssue(28961)]
```

Comment:
You should remove the ActiveIssue. It will disable the test on all platforms, not just Windows. 

You already added the PlatformSpecific attribute for Windows. So, that should be sufficient and will result in the test running on Windows at least.  For reference you could have the following comment:

```c#
[PlatformSpecific(TestPlatforms.Windows)] // NTLM support required. See: https://github.com/dotnet/corefx/issues/28961
```